Sources/BrowserServicesKit/SecureVault/AutofillSecureVault.swift

```diff
@@ -59,39 +73,31 @@ public protocol SecureVault {
     @discardableResult
     func storeCreditCard(_ card: SecureVaultModels.CreditCard) throws -> Int64
     func deleteCreditCardFor(cardId: Int64) throws
-}
-
-/// Protocols can't be nested, but classes can.  This struct provides a 'namespace' for the default implementations of the providers to keep it clean for other things going on in this library.
-internal struct SecureVaultProviders {
-
-    var crypto: SecureVaultCryptoProvider
-    var database: SecureVaultDatabaseProvider
-    var keystore: SecureVaultKeyStoreProvider
 
 }
 
-class DefaultSecureVault: SecureVault {
+public class DefaultAutofillSecureVault<T: AutofillDatabaseProvider>: AutofillSecureVault {
+
+    public typealias AutofillStorageProviders = SecureStorageProviders<T>
 
     private let lock = NSLock()
-    private let queue = DispatchQueue(label: "Secure Vault")
 
-    private let providers: SecureVaultProviders
+    private let providers: AutofillStorageProviders
     private let expiringPassword: ExpiringValue<Data>
 
-    var authExpiry: TimeInterval {
+    public var authExpiry: TimeInterval {
         return expiringPassword.expiresAfter
     }
```

Comment:
The concept of auth expiry has been left alone in the Autofill vault in the end, but it's been totally removed from the generic layer above.